src/sentry_plugins/heroku/plugin.py

```diff
@@ -25,12 +31,28 @@ def get_auth(self):
     def get_client(self):
         return HerokuApiClient()
 
+    def is_valid_signature(self, body, heroku_hmac):
+        secret = ProjectOption.objects.get_value(project=self.project, key="heroku:webhook_secret")
+        computed_hmac = base64.b64encode(
+            hmac.new(
+                key=secret.encode("utf-8"),
+                msg=body.encode("utf-8"),
+                digestmod=sha256,
+            ).digest()
+        ).decode("utf-8")
+
+        return hmac.compare_digest(heroku_hmac, computed_hmac)
+
     def handle(self, request: Request) -> Response:
-        email = None
-        if "user" in request.POST:
-            email = request.POST["user"]
-        elif "actor" in request.POST:
-            email = request.POST["actor"].get("email")
+        heroku_hmac = request.headers.get("Heroku-Webhook-Hmac-SHA256")
+        if not self.is_valid_signature(request.body.decode("utf-8"), heroku_hmac):
+            logger.error("heroku.webhook.invalid-signature", extra={"project_id": self.project.id})
+            return HttpResponse(status=401)
```

Comment:
Not sure if a human calls this endpoint or not, but does it make sense to add some extra context to the response body like:
https://github.com/getsentry/sentry/blob/52091f0f5e425c6abf788d092de690b7337b5801/src/sentry/web/frontend/release_webhook.py#L113-L118

Reply:
I don't think we expect a human to call this, when you deploy your Heroku code this automatically gets triggered.